nimSHA2.nim

```diff
@@ -248,8 +247,8 @@ proc update*(ctx: var SHA224, input: string) =
     if (ctx.count[0] and 0x3F) == 0:
       transform256(ctx.state, cast[cstring](addr(ctx.buffer[0])))
 
-proc update*(ctx: var SHA256, input: string) {.inline.} =
-  SHA224(ctx).update(input)
+func update*(ctx: var (SHA224|SHA256), input: string)  {.inline.} =
+  update(ctx, cast[seq[byte]](input))
```

Comment:
I would prefer avoid using cast here because we can use generic openarray to accept `string`, `seq`, and `array`.
example:
```Nim
func update*[T:char|byte](ctx: var (SHA224|SHA256), data: openarray[T]) =
   ...
```

thus we can remove the `func update*(ctx: var (SHA224|SHA256), input: string)  {.inline.} =`

Reply:
The problem is `string|openarray[byte]` doesn't compile to something that takes `array[N,byte]`, this is a bug that goes back as long as I can remember.

Reply:
this snippet below works with 1.2.12, which one doesn't compile, can you provide example?

```Nim
proc update*[T:char|byte](x: openArray[T]) =
  debugEcho x.len
  
var a: array[10, byte]
update(a)

var b: array[10, char]
update(b)

var c = newSeq[byte](10)
update(c)

var d = newSeq[char](10)
update(d)

update("hello")
```

Reply:
Thank you, I was trying `proc update*(x: string|openArray[byte])` but now I know better.